src/DevBetterWeb.Core/ValueObjects/BillingDetails.cs

```diff
@@ -1,24 +1,115 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using CSharpFunctionalExtensions;
+using DevBetterWeb.Core.Enums;
+using System;
 
 namespace DevBetterWeb.Core.ValueObjects
 {
   public class BillingDetails : ValueObject
   {
-    public string Message { get; }
     public decimal Amount { get; }
+    public string MemberName { get; }
+    public string SubscriptionPlanName { get; }
+    public string ActionVerbPastTense { get; }
+    public BillingPeriod BillingPeriod { get; }
+    public DateTime Date { get; }
 
-    public BillingDetails(string message, decimal amount = 0)
+    public BillingDetails(string memberName, string subscriptionPlanName, string actionVerbPastTense, BillingPeriod billingPeriod, DateTime date, decimal amount = 0)
     {
-      Message = message;
       Amount = amount;
+      MemberName = memberName;
+      SubscriptionPlanName = subscriptionPlanName;
+      ActionVerbPastTense = actionVerbPastTense;
+      BillingPeriod = billingPeriod;
+      Date = date;
+    }
+
+    public string GetMessageForMemberView()
+    {
+      string message = GetSentenceWithProperSyntax(Viewer.Member);
+
+      return message;
+    }
+
+    public string GetMessageForAdminView(string memberId)
+    {
+      string message = GetSentenceWithProperSyntax(Viewer.Admin, memberId);
+      return message;
+    }
+
+    private string GetSentenceWithProperSyntax(Viewer viewer, string memberId = "")
+    {
+      string message = "";
+
+      switch(viewer)
+      {
+        case Viewer.Admin:
+          message += $"Member ID: {memberId}. {MemberName}";
+          break;
+        case Viewer.Member:
+          message += $"You";
+          break;
+        default:
+          break;
+      }
+
+      switch (ActionVerbPastTense)
```

Comment:
Ugh I hate this. Last thing I wanted when introducing the verb variable was to need a switch. I can see why you went this way, though.

I don't have a better solution at the moment. And at least you should be able to write unit tests for this one easily enough.